src/prefect/orion/models/deployments.py

```diff
@@ -140,7 +141,19 @@ async def update_deployment(
 
     # exclude_unset=True allows us to only update values provided by
     # the user, ignoring any defaults on the model
-    update_data = deployment.dict(shallow=True, exclude_unset=True)
+    update_data = deployment.dict(
+        shallow=True,
+        exclude_unset=True,
+        exclude={"worker_pool_name", "worker_pool_queue_name"},
+    )
+    if deployment.worker_pool_name and deployment.worker_pool_queue_name:
```

Comment:
same comment here - shouldn't this only check `pool_name`?

Reply:
I added logic to handle just a pool name and use the default queue here also.